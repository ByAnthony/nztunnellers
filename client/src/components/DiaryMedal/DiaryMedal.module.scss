@use 'src/index.scss';

.other-medal {
  font-size: 1rem;

  @include index.fullwidth-main-card;

  border-bottom: 1px solid index.$primary-light-color;
  background-color: index.$primary-dark-color;

  @include index.tablet {
    font-size: 1.1rem;
  }

  @include index.desktop {
    font-size: 1.15rem;
  }

  @include index.large-desktop {
    font-size: 1.35rem;
  }

  p {
    margin-bottom: index.$xs-margin;
  }
}

.citation {
  margin-top: 5px;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.6rem;

  @include index.desktop {
    font-size: 1.1rem;
    line-height: 1.6rem;
  }

  @include index.large-desktop {
    font-size: 1.25rem;
    line-height: 2rem;
  }
}
