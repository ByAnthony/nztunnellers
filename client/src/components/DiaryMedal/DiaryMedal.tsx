import { Medal } from '../../types/tunneller';
import STYLES from '../ProfileDiary/ProfileDiary.module.scss';
import OTHER_MEDALS_STYLES from './DiaryMedal.module.scss';

type Props = {
    medals: Medal[] | [];
}

function BritishWarAndVictoryMedals({ medalsList }: {medalsList: Medal[] | []}) {
  const filteredMedals = medalsList.filter((medal) => medal.name === 'British War Medal' || medal.name === 'Victory Medal');

  if (filteredMedals.length > 0) {
    return (
      <div className={STYLES['halfwidth-cards-container']}>
        {filteredMedals.map((medal) => (
          <div key={filteredMedals.indexOf(medal)} className={STYLES['halfwidth-secondary-card']}>
            <p><img src={`/images/roll/medals/${medal.image}`} alt={`${medal.name} ribbon`} width={40} /></p>
            <span>{ medal.name }</span>
          </div>
        ))}
      </div>
    );
  }
  return null;
}

function OtherMedals({ medalsList }: {medalsList: Medal[] | []}) {
  const filteredMedals = medalsList.filter((medal) => medal.name !== 'British War Medal' && medal.name !== 'Victory Medal');
  return (
    <>
      {filteredMedals.map((medal) => {
        const displayCountry = (medal.country !== 'United Kingdom') ? `(${medal.country})` : '';
        return (
          <div key={filteredMedals.indexOf(medal)} className={OTHER_MEDALS_STYLES['other-medal']}>
            <p><img src={`/images/roll/medals/${medal.image}`} alt={`${medal.name} ribbon`} width={40} /></p>
            <span>{`${medal.name} ${displayCountry}`}</span>
            <div className={OTHER_MEDALS_STYLES.citation}><span>{`${medal.citation}.`}</span></div>
          </div>
        );
      })}
    </>
  );
}

export function DiaryMedal({ medals }: Props) {
  return (
    <>
      <BritishWarAndVictoryMedals medalsList={medals} />
      <OtherMedals medalsList={medals} />
    </>
  );
}
