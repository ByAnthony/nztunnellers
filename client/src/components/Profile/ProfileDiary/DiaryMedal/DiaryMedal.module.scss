@use 'src/index.scss';

.other-medal {
  font-size: 1rem;

  @include index.fullwidth-main-card;

  border-bottom: 1px solid index.$primary-light-color;
  background-color: index.$primary-dark-color;

  @include index.tablet {
    font-size: 1.1rem;
  }

  @include index.desktop {
    font-size: 1.15rem;
  }
}

.citation {
  margin-top: index.$xxs-margin;
  font-size: 1rem;
  font-weight: 200;
  line-height: 1.6rem;

  @include index.desktop {
    font-size: 1.1rem;
    line-height: 1.6rem;
  }
}
