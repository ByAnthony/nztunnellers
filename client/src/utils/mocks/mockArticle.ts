import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
