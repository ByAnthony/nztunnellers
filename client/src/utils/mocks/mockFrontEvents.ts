import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
