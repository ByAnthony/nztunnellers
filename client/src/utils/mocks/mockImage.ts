import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
