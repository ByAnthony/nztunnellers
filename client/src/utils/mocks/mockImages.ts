import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook = (overrides: Partial<ImageBook>
    | undefined = undefined): ImageBook => ({
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: '89',
  ...overrides,
});

const mockSource = (overrides: Partial<ImageSource> | undefined = undefined): ImageSource => ({
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
  ...overrides,
});

export const mockImages = (overrides: Partial<Image> | undefined = undefined): Image => ({
  url: '',
  source: mockSource(),
  ...overrides,
});
