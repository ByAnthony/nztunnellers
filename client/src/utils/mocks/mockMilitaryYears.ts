import {
  Date,
  DeathPlace,
  Death,
  DeathCause,
  Cemetery,
  Medal,
  MilitaryYears,
  Enlistment,
  EmbarkationUnit,
  Training,
  Transport,
  EndOfService,
  Demobilization,
} from '../../types/tunneller';
import { mockFrontEvents } from './mockFrontEvents';

export const mockBritishWarMedal: Medal = {
  name: 'British War Medal',
  country: 'United Kingdom',
  image: 'british-war-medal.png',
  citation: null,
};

export const mockVictoryMedal: Medal = {
  name: 'Victory Medal',
  country: 'United Kingdom',
  image: 'victory-medal.png',
  citation: null,
};

export const mockMedals: Medal[] = [
  mockBritishWarMedal,
  mockVictoryMedal,
];

export const mockDate: Date = {
  year: '1926',
  dayMonth: '13 October',
};

const mockPlace: DeathPlace = {
  location: 'Telegraph Hill',
  town: 'Arras',
  country: 'France',
};

const mockCause: DeathCause = {
  type: 'Killed in action',
  circumstances: 'Dead wounded by shell fire',
};

const mockCemetery: Cemetery = {
  name: 'Faubourg d\'Amiens',
  location: 'Arras',
  country: 'France',
  graveReference: 'IE 18',
};

export const mockDeath: Death = {
  date: mockDate,
  place: mockPlace,
  cause: mockCause,
  cemetery: mockCemetery,
  ageAtDeath: 89,
};

export const mockEnlistment: Enlistment = {
  serial: '1/1000',
  rank: 'Sapper',
  date: mockDate,
  district: 'Canterbury',
  alias: null,
  transferredToTunnellers: null,
};

const mockTraining: Training = {
  date: mockDate,
  location: 'Auckland',
  locationType: 'Military Camp',
};

export const mockEmbarkationUnit: EmbarkationUnit = {
  detachment: 'Main Body',
  training: mockTraining,
  section: 'Section No.2',
  attachedCorps: null,
};

export const mockTransport: Transport = {
  reference: 'HMNZT 56',
  vessel: 'Beata',
  departureDate: mockDate,
  departurePort: 'Auckland',
  arrivalDate: mockDate,
  arrivalPort: 'Plymouth',
};

const mockDemobilization: Demobilization = {
  date: mockDate,
  country: 'New Zealand',
};

const mockEndOfService: EndOfService = {
  deserter: false,
  transferred: null,
  deathWar: null,
  transportNz: mockTransport,
  demobilization: mockDemobilization,
};

export const mockMilitaryYears: MilitaryYears = {
  enlistment: mockEnlistment,
  embarkationUnit: mockEmbarkationUnit,
  transportUk: mockTransport,
  frontEvents: mockFrontEvents,
  endOfService: mockEndOfService,
  medals: mockMedals,
};
