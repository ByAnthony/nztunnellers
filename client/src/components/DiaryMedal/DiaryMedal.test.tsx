import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { mockMedals } from '../../utils/mocks/mockMilitaryYears';
import { DiaryMedal } from './DiaryMedal';

const component = (<DiaryMedal medals={mockMedals} />);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});
