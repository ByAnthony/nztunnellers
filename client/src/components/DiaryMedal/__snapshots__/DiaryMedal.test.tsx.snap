// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="halfwidth-cards-container"
  >
    <div
      class="halfwidth-secondary-card"
    >
      <p>
        <img
          alt="British War Medal ribbon"
          src="/images/roll/medals/british-war-medal.png"
          width="40"
        />
      </p>
      <span>
        British War Medal
      </span>
    </div>
    <div
      class="halfwidth-secondary-card"
    >
      <p>
        <img
          alt="Victory Medal ribbon"
          src="/images/roll/medals/victory-medal.png"
          width="40"
        />
      </p>
      <span>
        Victory Medal
      </span>
    </div>
  </div>
</DocumentFragment>
`;
